src/mono/wasm/runtime/binding_support.js

```diff
@@ -2042,9 +1987,9 @@ var BindingSupportLib = {
 				};
 
 				var res = allocator(coreObj, js_args);
-				var gc_handle = BINDING.mono_wasm_free_list.length ? BINDING.mono_wasm_free_list.pop() : BINDING.mono_wasm_ref_counter++;
-				BINDING.mono_wasm_object_registry[gc_handle] = res;
-				return BINDING.mono_wasm_convert_return_value(gc_handle + 1);
```

Comment:
Good job catching this bug, do we have test coverage for this path? If the return value isn't being used and that was why the bug didn't get caught before, we should remove the return value. Otherwise, it needs a test

Reply:
This was not bug, just bad name. It was very confusing to me.

Reply:
it was ```gc_handle + 1``` before and now it's ```js_handle```, the + 1 looked like a bug so I assume that's why it's gone?

Reply:
That math was confusing. I _think_ that it tried to not send zero handle to C# side for `SafeHandle`'s sake.